elyra/metadata/schemas/kfp.json

```diff
@@ -47,7 +47,7 @@
           "title": "Kubeflow Pipelines API Endpoint",
           "description": "The Kubeflow Pipelines API endpoint",
           "type": "string",
-          "format": "uri",
+          "pattern": "(?:(?:http|https)://)?(www\\.)?[a-zA-Z0-9@:%._\\+~#?&//=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%._\\+~#?&//=]*)",
```

Comment:
While this allows schema less URLs to be accepted, no error is thrown when the input doesn't match the regexp.

Reply:
Wow, that's quite an regex. What about stealing some ideas from https://www.stephenlewis.me/blog/json-schema-url-validation/? They rely on format/hostname in json schema spec, https://json-schema.org/understanding-json-schema/reference/string#hostnames, plus the `anyOf` operator.

When I leave protocol out, does that imply `https`, or is it `http or https`?

Can this be tested here? https://github.com/opendatahub-io/elyra/blob/3f11b019939effbac016862f8caffda8f560e724/elyra/tests/cli/resources/runtime_configs/valid_kfp_test_config.json#L4

Reply:
Thanks for sharing this @jiridanek . While this is indeed a much cleaner way to perform the validation. It am not able to get through scenario of `.com/v1` or `.com/`. Let me check if I can modify the pattern to handle these cases as well.

Reply:
> It am not able to get through scenario of .com/v1 or .com/.

The article actually said at the end that with their approach this does not work. I missed it when I was reading it ;(

> ## Scheme-less URLs
>
> But what about “scheme-less” URLs, such as www.example.com?
>
> This is where JSON Schema comes up short. If you’re dealing with just a domain, and no path, the following definition covers the basics. It fails miserably on something like www.example.com/endpoint, though.
>
> ...
>
> Either way, you’ll still need to do some manual validation.